modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/service/impl/DDMTemplateLocalServiceImpl.java

```diff
@@ -1919,6 +1938,16 @@ private long[] _getAncestorSiteAndDepotGroupIds(long groupId) {
 		}
 	}
 
+	private boolean _isValidImageURL(String smallImageURL) {
+		String regex = "(http(s?):/)(/[^/]+)+.(?:jpg|jpeg|gif|png)";
+
+		Pattern pattern = Pattern.compile(regex);
+
+		Matcher matcher = pattern.matcher(smallImageURL);
+
+		return matcher.find();
+	}
```

Comment:
No, because the isUrl method don't validate a image Url specifically, just a generic Url.

Reply:
@MrJohn1911 Can you guarantee that all valid URLs will be accepted by this code? What if the user tries to use a short link or a different (but valid) extension?

Reply:
I think i can't handle the use of a short link, but i can add new valid image extensions, like .svg.

Reply:
> I think i can't handle the use of a short link, 

@MrJohn1911  I suggest talking to Luiz and Julia about the expected behavior of this bug. In my opinion, it would be enough to just validate the URL using `Validator.isUrl` and not try to validate its content.

Reply:
I talked with Julia Lira and she said that i have to validate if the url is a image url, but she doesn't know if it causes a security problem.

Reply:
@MrJohn1911 in this case, could you please take a look if there are other ways to validate the url? 

Maybe the following link can help you: https://stackoverflow.com/questions/15293339/how-to-know-if-the-image-exists-or-not-by-reading-url-in-java

Reply:
@carolmariaabb I checked it today and I changed the method to look like this:

```
private boolean _isInvalidImageURL(String smallImageURL) {
   try {
      ImageIO.read(new URL(smallImageURL));

      return false;
   }
   catch (IOException ioException) {
      if (_log.isDebugEnabled()) {
         _log.debug(ioException);
      }
   }

   return true;
}
```
Did you think that this has any problem? I tested it with an valid URL, an invalid URL and even with a short link of an image URL and it works!

Reply:
@MrJohn1911 looks good to me